src/main/java/fi/fmi/avi/model/taf/immutable/TAFImpl.java

```diff
@@ -178,6 +181,257 @@ public Builder withCompleteForecastTimes(final ZonedDateTime reference) {
+                super.setReportStatus(ReportStatus.NORMAL).setMissingMessage(true).setCancelMessage(false);
+            } else {
+                if (TAFStatus.NORMAL.equals(status)) {
+                    super.setReportStatus(ReportStatus.NORMAL);
+                } else if (TAFStatus.AMENDMENT.equals(status)) {
+                    super.setReportStatus(ReportStatus.AMENDMENT);
+                } else if (TAFStatus.CORRECTION.equals(status)) {
+                    super.setReportStatus(ReportStatus.CORRECTION);
+                } else {
+                    throw new IllegalArgumentException("Cannot determine report status from TAFStatus " + status);
+                }
+                super.setMissingMessage(false).setCancelMessage(false);
+            }
+            return this;
+        }
+
+        /**
+         * Provides the current builder value of the status property.
+         *
+         * @return the message status
+         * @deprecated migrate to using a combination of {@link #getReportStatus()}, {@link #isCancelMessage()} and
+         * * {@link #isMissingMessage()} instead
+         */
+        @Override
+        @Deprecated
+        public TAFStatus getStatus() {
+            return super.getStatus();
+        }
+
+        /**
+         * Sets the report message status.
+         * <p>
+         * Note that for backward compatibility this method also sets the <code>status</code> property with the following logic:
+         * <ul>
+         *     <li>status is {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#NORMAL}: <code>status =</code>
+         *     {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#NORMAL}</li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#AMENDMENT}: <code>status =</code>
+         *     {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#AMENDMENT}</li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#CORRECTION}: <code>status =</code>
+         *     {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#CORRECTION}</li>
+         * </ul>
+         *
+         * @param status the message status to set
+         * @return the builder
+         * @see #setReportStatus(ReportStatus)
+         */
+        @Override
+        public Builder setReportStatus(final ReportStatus status) {
+            super.setReportStatus(status);
+            if (ReportStatus.NORMAL.equals(status)) {
+                super.setStatus(TAFStatus.NORMAL);
+            } else if (ReportStatus.AMENDMENT.equals(status)) {
+                super.setStatus(TAFStatus.AMENDMENT);
+            } else if (ReportStatus.CORRECTION.equals(status)) {
+                super.setStatus(TAFStatus.CORRECTION);
+            } else {
+                throw new IllegalArgumentException("Cannot determine TAFStatus from report status " + status);
+            }
+            return this;
+        }
```

Comment:
The example above assumes `getReportStatus()` returns non-optional `ReportStatus`. At least yet the return value is `Optional<ReportStatus>`, `getReportStatus().orElse(ReportStatus.NORMAL)` should be sufficient to adapt.

Reply:
It might be a good idea to add test cases like the example at top of the [comment above](https://github.com/fmidev/fmi-avi-messageconverter/pull/68#discussion_r523933303) to the `TAFIWXXM21_30_ResilienceTest`.

Reply:
Added the resilience test to cover this, and implemented the migration code to TAFStatus as suggested